Makefile

```diff
@@ -107,15 +116,38 @@ clientset: ## Generate a typed clientset
 clean: ## Remove all generated files
 	rm -f bazel-*
 
+.PHONY: all-docker-build
+all-docker-build: $(addprefix sub-docker-build-,$(ALL_ARCH))
+	@echo "updating kustomize image patch file for manager resource"
+	hack/sed.sh -i.tmp -e 's@image: .*@image: '"$(CONTROLLER_IMG):$(TAG)"'@' ./config/default/manager_image_patch.yaml
+
+sub-docker-build-%:
+	$(MAKE) ARCH=$* docker-build
+
 .PHONY: docker-build
 docker-build: generate fmt vet manifests ## Build the docker image for controller-manager
-	docker build . -t ${CONTROLLER_IMG}
+	docker build --build-arg ARCH=$(ARCH) . -t $(CONTROLLER_IMG)-$(ARCH):$(TAG)
 	@echo "updating kustomize image patch file for manager resource"
-	hack/sed.sh -i.tmp -e 's@image: .*@image: '"${CONTROLLER_IMG}"'@' ./config/default/manager_image_patch.yaml
+	hack/sed.sh -i.tmp -e 's@image: .*@image: '"${CONTROLLER_IMG}-$(ARCH):$(TAG)"'@' ./config/default/manager_image_patch.yaml
+
+.PHONY:all-push ## Push all the architecture docker images and fat manifest docker image
+all-push: all-docker-push push-manifest
+
+.PHONY:all-docker-push ## Push all the architecture docker images
+all-docker-push: $(addprefix sub-docker-push-,$(ALL_ARCH))
+
+sub-docker-push-%:
+	$(MAKE) ARCH=$* docker-push
+
+.PHONY: push-manifest
+push-manifest: ## Push the fat manifest docker image
+	docker manifest create --amend $(CONTROLLER_IMG):$(TAG) $(shell echo $(ALL_ARCH) | sed -e "s~[^ ]*~$(CONTROLLER_IMG)\-&:$(TAG)~g")
```

Comment:
Would it make sense to use `manifest-tool` here instead of `docker manifest` to avoid requiring a minimum version of `docker`?

Reply:
As we don't see lot activities in the `manifest-tool` projects, its better to use `docker manifest` command instead as we have adopted docker command in most of the places in `k/k` repo.

Reply:
Sounds good. In that case, lets go ahead and document the minimum version of docker somewhere.

Reply:
I will following note:

```
Minimum docker version 18.06.0 is required for creating and pushing manifest images
```

Though manifest command enabled much this release but because of some known issues before, commad is stable post `18.06.0`